src/bp/core/repositories/conversations.ts

```diff
@@ -0,0 +1,238 @@
+import * as sdk from 'botpress/sdk'
+import Database from 'core/database'
+import { TYPES } from 'core/types'
+import { inject, injectable } from 'inversify'
+import _ from 'lodash'
+import moment from 'moment'
+import ms from 'ms'
+import uuid from 'uuid'
+
+const conversationsTable = 'web_conversations'
+const messageTable = 'web_messages'
+
+@injectable()
+export class ConversationsRepository {
+  constructor(@inject(TYPES.Database) private database: Database) {}
+
+  async getOrCreateConversation(
+    botId: string,
+    userId: string,
+    opt: { lifetime?: string; isUserCreated?: boolean } = { lifetime: '6 hours', isUserCreated: false }
+  ): Promise<any> {
+    const lessThanLifetime = this.database.knex.date.isAfter(
+      'last_heard_on',
+      moment()
+        .subtract(ms(opt.lifetime), 'ms')
+        .toDate()
+    )
+
+    const conversation = await this.database
+      .knex(conversationsTable)
+      .select('id')
+      .whereNotNull('last_heard_on')
+      .andWhere({ userId, botId })
+      .andWhere(lessThanLifetime)
+      .orderBy('last_heard_on', 'desc')
+      .limit(1)
+      .get(0)
+
+    return conversation ? conversation['id'] : this.createConversation(botId, userId, opt.isUserCreated)
+  }
+
+  async createConversation(botId: string, userId: string, isCreatedByUser = false): Promise<any> {
+    const uid = Math.random()
+      .toString()
+      .substr(2, 6)
+    const title = `Conversation ${uid}`
+
+    await this.database.knex(conversationsTable).insert({
+      botId,
+      userId,
+      created_on: this.database.knex.date.now(),
+      last_heard_on: isCreatedByUser ? this.database.knex.date.now() : undefined,
+      title
+    })
+
+    const conversation = await this.database
+      .knex(conversationsTable)
+      .where({ title, userId, botId })
+      .select('id')
+      .get(0)
+
+    return conversation && conversation['id']
+  }
+
+  async appendUserMessage(
+    botId: string,
+    userId: string,
+    conversationId: string,
+    fullName: string,
+    avatarUrl: string,
+    payload: sdk.conversations.MessagePayload
+  ) {
+    const convo = await this.database
+      .knex(conversationsTable)
+      .where({ userId, id: conversationId, botId })
+      .select('id')
+      .limit(1)
+      .get(0)
+
+    if (!convo) {
+      throw new Error(`Conversation "${conversationId}" not found`)
+    }
+
+    const { type, text, raw, data } = payload
+
+    const message = {
+      id: uuid.v4(),
```

Comment:
2 simple questions

 * Why do we both use uuid and nanoid libraries ?
 * Why not use incremental SQL sequences  for ids ?